exec/java-exec/src/main/java/org/apache/drill/exec/physical/impl/project/ProjectRecordBatch.java

```diff
@@ -136,6 +145,10 @@ public VectorContainer getOutgoingContainer() {
 
   @Override
   protected IterOutcome doWork() {
+    if (wasNone) {
+      return IterOutcome.NONE;
+    }
+
     int incomingRecordCount = incoming.getRecordCount();
 
     if (first && incomingRecordCount == 0) {
```

Comment:
Actually, if the first batch was non-empty, the new changes wouldn't apply because of the following check: 
      if (first && incomingRecordCount == 0) { ... }
Then if the next incoming  batch is empty, it should continue to work since we have already produced the schema from the first batch.  On the other hand if the first batch is empty and we see a NONE iterator outcome, we want to make sure that a schema is produced but at the same time not call next() since a NONE outcome has already been seen.